Libraries/PeriphDrivers/Source/TMR/tmr_revb.c

```diff
@@ -276,12 +278,12 @@ int MXC_TMR_RevB_SetPWM(mxc_tmr_revb_regs_t *tmr, uint32_t pwm)
         MXC_TMR_RevB_ClearFlags(tmr); // Clear flags so we can catch the next one
         while (!MXC_TMR_RevB_GetFlags(tmr)) {} // Wait for next PWM transition
         MXC_TMR_RevB_Stop(tmr); // Pause timer
-        MXC_TMR_RevB_SetCount(tmr, 0); // Reset the count
+        MXC_TMR_RevB_SetCount(tmr, 1); // Reset the count
```

Comment:
What is the desired behavior for PWM == 0?  Do we still want a very small signal in this case?

For https://github.com/Analog-Devices-MSDK/msdk/pull/818  I assumed the desired behavior for a PWM of 0 is no signal (0% duty cycle = no transitions), so I may have been mistaken.

Reply:
By assuming polarity is 1 as defined below picture.
![pwm_polarity](https://github.com/Analog-Devices-MSDK/msdk/assets/94184469/d16713b8-c17a-436a-8982-1ced17681a5e)

Desired behavior for PWM == 0 is continues inactive signal.
![pwm_behavior](https://github.com/Analog-Devices-MSDK/msdk/assets/94184469/6c04d6bd-9a81-4e47-b5fe-cf788854b241)

However, if PWM is set to 0, the output signal is observed like following picture. Because of cnt can not match with pwm value.
![pwm_behavior_wrong](https://github.com/Analog-Devices-MSDK/msdk/assets/94184469/1e8afc48-47ce-4ab9-ba50-a2bb150beed9)

If program changes 0 pwm values to 1 like this code does `tmr->pwm = pwm ? pwm : 1;` then output signal act like following picture.
![pwm_1](https://github.com/Analog-Devices-MSDK/msdk/assets/94184469/7d2d0bcf-df7d-449e-b24f-7c8fe384c5ee)

I couldn't find a way to fix this small pulse without a workaround because of timer resets cnt register to 0x1 instead of 0x0. But at least this solution gives %0 duty cycle output signal.

Reply:
> Desired behavior for PWM == 0 is continues inactive signal.

For polarity == 1 the _inactive_ signal is logic high.  So I think that pwm = 0 achieves the desired behavior.  

But in the second image the polarity is 1.  For polarity == 1 the inactive signal is always _high_.  To achieve the desired behavior with pwm == 0, can't the polarity just be set to 0?  Then the inactive signal will always be low, while still achieving the desired result for `pwm=period/2` and `pwm=period` tests.

My concern with overwriting to pwm = 1 is the little pulses in the waveform, so it's a non-zero duty cycle.

Reply:
> What is the desired behavior for PWM == 0? Do we still want a very small signal in this case?
> 
> For #818 I assumed the desired behavior for a PWM of 0 is no signal (0% duty cycle = no transitions), so I may have been mistaken.

Desired signal for pwm==0 is no signal, we handled this case by upper layer before calling sdk driver, see below links.
So that, @hfakkiz  will double check pwm=pwm?pwm:1 and MXC_TMR_RevB_SetCount(tmr, 1), if this changes are not necessary for our case, please remove them.

https://github.com/adi-innersource/zephyr/blob/MAX32xxx-Driver-PWM/drivers/pwm/pwm_max32.c
![image](https://github.com/Analog-Devices-MSDK/msdk/assets/46590392/88b58968-4d00-4fb8-a060-59d0b5c9fa0e)

Reply:
Removed related commit.